File: backend/src/main/java/hu/unideb/inf/webfejlesztesbeadando/controller/SiteUserController.java
package hu.unideb.inf.webfejlesztesbeadando.controller;

import hu.unideb.inf.webfejlesztesbeadando.DTO.OrderDTO;
import hu.unideb.inf.webfejlesztesbeadando.DTO.SiteUserDTO;
import hu.unideb.inf.webfejlesztesbeadando.service.OrderService;
import hu.unideb.inf.webfejlesztesbeadando.service.SiteUserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@CrossOrigin(origins = "http://localhost:3000")
@RequestMapping("/siteUsers")
public class SiteUserController {

    @Autowired
    SiteUserService siteUserService;

    @GetMapping("/getAll")
    ResponseEntity getAllOrders() {
        return new ResponseEntity<>(siteUserService.readAll(), HttpStatus.OK);
    }

    @PostMapping("/createUser")
    ResponseEntity createOrder(@RequestBody SiteUserDTO siteUserDTO) {
        return new ResponseEntity(siteUserService.create(siteUserDTO), HttpStatus.CREATED);
    }

    @PostMapping("/updateUser")
    ResponseEntity updateOrder(@RequestBody SiteUserDTO siteUserDTO) {
        return new ResponseEntity(siteUserService.update(siteUserDTO), HttpStatus.OK);
    }

    @PostMapping("/deleteUser")
    ResponseEntity deleteOrder(@RequestBody Long id) {
        siteUserService.delete(id);
        return new ResponseEntity(HttpStatus.OK);
    }

}

File: frontend/src/controller/siteUserController.js
import axios from "axios"
import { baseURL } from "./constants"

export class SiteUserController {
    static async getAll() {
        const respone = await axios.get(`${baseURL}/siteUsers/getAll`)
        return respone.data
    }

    static async createSiteUser(siteUserDTO) {
        const respone = await axios.post(`${baseURL}/siteUsers/createUser`, siteUserDTO)
        return respone.data
    }


    static async updateSiteUser(siteUserDTO) {
        const respone = await axios.patch(`${baseURL}/siteUsers/updateUser`, siteUserDTO)
        return respone.data
    }


    static async deleteSiteUser(siteUserDTO) {
        const respone = await axios.delete(`${baseURL}/siteUsers/deleteUser`, {
            data: siteUserDTO
        })
        return respone.data
    }
}